src/api/repositories/user/user_repository_mock.go

```diff
@@ -0,0 +1,35 @@
+package repositories
+
+import (
+	"github.com/fullstacktf/personal-nutritionist-backend/api/models"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/mock"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type UserRepositoryMock struct {
+	mock.Mock
+}
+
+func (m *UserRepositoryMock) GetUsers(c *gin.Context) ([]models.User, error) {
+	args := m.Called(c)
+	return args.Get(0).([]models.User), nil
```

Comment:
We should return args.Get(1) like error so we can mock the error case

Reply:
Remind! It returns the values we set in the return parameters `.... .Return(usersMock, nil)`in tests.